ratelimit.go

```diff
@@ -24,35 +25,39 @@ func CreateUnlimitedRateLimiter() *RateLimiter {
 }
 
 func createRateLimiter(limit int) *RateLimiter {
+	var window []time.Time
+
 	return &RateLimiter{
 		joinLimit: limit,
-		throttle:  make(chan time.Time, 10),
+		window:    window,
 	}
 }
 
-func (r *RateLimiter) Throttle() {
+func (r *RateLimiter) Throttle(count int) {
```

Comment:
Should there be a mutex on this, or is it handled before the call?

Reply:
The writer runs in a channel so there should be not concurrent access of this function